src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2352,51 +2409,103 @@ private void ReportNullabilityMismatchWithTargetDelegate(SyntaxNode syntax, Name
+            bool checkConversion,
+            bool fromExplicitCast,
+            out bool canConvert)
+        {
+            Debug.Assert(node != null);
+            Debug.Assert(operandOpt != null || (object)operandType != null);
+            Debug.Assert((object)targetType != null);
+
             bool? isNullableIfReferenceType = null;
+            canConvert = true;
 
             switch (conversion.Kind)
             {
                 case ConversionKind.MethodGroup:
+                    if (!fromExplicitCast)
+                    {
+                        ReportNullabilityMismatchWithTargetDelegate(operandOpt.Syntax, targetType.GetDelegateType(), conversion.Method);
+                    }
+                    isNullableIfReferenceType = false;
+                    break;
+
                 case ConversionKind.AnonymousFunction:
                 case ConversionKind.InterpolatedString:
                     isNullableIfReferenceType = false;
                     break;
 
                 case ConversionKind.ExplicitUserDefined:
                 case ConversionKind.ImplicitUserDefined:
-                    var methodOpt = conversion.Method;
-                    if ((object)methodOpt != null && methodOpt.ParameterCount == 1)
+                    // cf. Binder.CreateUserDefinedConversion
                     {
-                        if (conversion.BestUserDefinedConversionAnalysis.Kind == UserDefinedConversionAnalysisKind.ApplicableInLiftedForm)
+                        if (!conversion.IsValid)
                         {
-                            // PROTOTYPE(NullableReferenceTypes): Conversions: Lifted operator
+                            canConvert = false;
                             break;
                         }
-                        // PROTOTYPE(NullableReferenceTypes): Update method based on operandType.
-                        var resultType = methodOpt.ReturnType;
-                        // Ensure converted type matches expected target type, specifically for lifted
-                        // conversions. See LocalRewriter.MakeConversionNode for similar handling
-                        // based on conversion.BestUserDefinedConversionAnalysis.ToType.
-                        if (!fromConversionNode)
-                        {
-                            if (resultType.TypeSymbol != targetType)
-                            {
-                                // PROTOTYPE(NullableReferenceTypes): Should handle nested nullability
-                                // and top-level nullability. Conversion from resultType to targetType
-                                // should be treated as a built-in conversion.
-                                resultType = TypeSymbolWithAnnotations.Create(targetType);
-                            }
-                        }
-                        return resultType;
+
+                        // operand -> conversion "from" type
+                        operandType = ApplyConversion(
```

Comment:
The quoted "from" is confusing. It seems to have different meaning than just 'from', but unclear how so.

Reply:
The terms are from `Binder.CreateUserDefinedConversion`. Will leave as is for now since I suspect this block of code will change.

---
In reply to: [189379132](https://github.com/dotnet/roslyn/pull/26356#discussion_r189379132) [](ancestors = 189379132)